odoo/addons/base/tests/test_qweb.py

```diff
@@ -1762,6 +1702,977 @@ def test_space_remove_technical_all(self):
+                    <span>1</span>
+                    <article>
+                        <span>1</span>
+                    </article>
+                </section>
+            </root>
+        """
+        self.assertEqual(etree.fromstring(render), etree.fromstring(result), 'rendering 1')
+
+        render = IrQweb._render(template_page.id, {
+            'key1': (1,),
+            'key2': (1,),
+            'val': 2,
+        })
+        result = """
+            <root>
+                <section>
+                    <span>1</span>
+                    <article>
+                        <span>2</span>
+                    </article>
+                </section>
+            </root>
+        """
+        self.assertEqual(etree.fromstring(render), etree.fromstring(result), 'rendering 2')
+
+        render = IrQweb._render(template_page.id, {
+            'key1': (1,),
+            'key2': (2,),
+            'val': 3,
+        })
+        result = """
+            <root>
+                <section>
+                    <span>1</span>
+                    <article>
+                        <span>3</span>
+                    </article>
+                </section>
+            </root>
+        """
+        self.assertEqual(etree.fromstring(render), etree.fromstring(result), 'rendering 3')
+
+    def test_15_render_xml_conditional_cache(self):
+        view1 = self.env['ir.ui.view'].create({
+            'name': "dummy",
+            'type': 'qweb',
+            'arch': """
+                <t t-name="base.dummy">
+                    <div t-cache="(cache_id,) if condition else None" class="toto">
+                        <table>
+                            <tr><td><span t-esc="value[0]"/></td></tr>
+                            <tr><td><span t-esc="value[1]"/></td></tr>
+                            <tr><td><span t-esc="value[2]"/></td></tr>
+                        </table>
+                    </div>
+                </t>
+            """
+        })
+        expected_result = etree.fromstring("""
+            <div class="toto">
+                <table>
+                    <tr><td><span>1</span></td></tr>
+                    <tr><td><span>2</span></td></tr>
+                    <tr><td><span>3</span></td></tr>
+                </table>
+            </div>
+        """)
+
+        IrQweb = self.env['ir.qweb'].with_context(use_qweb_t_cache=True)
+
+        result = etree.fromstring(IrQweb._render(view1.id, {'cache_id': 1, 'condition': True, 'value': [1, 2, 3]}))
+        self.assertEqual(result, expected_result, 'First rendering (add in cache)')
+
+        result = etree.fromstring(IrQweb._render(view1.id, {'cache_id': 1, 'condition': True, 'value': [10, 20, 30]}))
+        self.assertEqual(result, expected_result, 'Next rendering use cache')
+
+        result = etree.fromstring(IrQweb._render(view1.id, {'cache_id': 1, 'value': [10, 20, 30]}))
+        self.assertNotEqual(result, expected_result, 'Next rendering use cache')
```

Comment:
I find this weird: in this case condition is None, so t-cache=None in the template, but it still renders the cached values.